napari/layers/utils/plane.py

```diff
@@ -0,0 +1,166 @@
+from typing import Tuple
+
+import numpy as np
+from pydantic import validator
+
+from ...utils.events import EventedModel, SelectableEventedList
+from ...utils.geometry import intersect_line_with_plane_3d
+from ...utils.translations import trans
+
+
+class Plane(EventedModel):
+    """Defines a plane in 3D (with optional thickness) and provides utility methods
+    and events to handle it.
+
+    A plane is defined by a position, a normal vector and a thickness value, and can
+    be enabled or disabled.
+
+    Attributes
+    ----------
+    position : 3-tuple
+        A 3D position on the plane, defined in data coordinates.
+    normal : 3-tuple
+        A 3D unit vector normal to the plane, defined in data coordinates.
+    thickness : float
```

Comment:
Maybe say something about a thickness of zero being used for slicing planes

Reply:
You probably mean clipping planes? I would not put it here, because
1) the plane is agnostic about what it's used for
2) thickness is actually unused by clipping planes, so it could be set to anything.
3) We discussed that we could maybe make clipping planes behave as a soft-edge mask when thickness is above 0.

If anything, I'd rather mention this in the `Image` and `Label` layer docstrings about clipping planes, saying that thickness is unused. What do you think?

Reply:
see point below about subclassing specific types of planes to simplify/remove the need for these sorts of discussions